peps/pep-0735.rst

```diff
@@ -0,0 +1,613 @@
+cross-compatibility between tools and will onboard beginning users into
+``pyproject.toml`` data more quickly (before they learn about the
+``build-system`` table, for example). It will also resolve the long-standing
+ambiguity and tension within the Python community about whether or not extras
+should be used to declare development dependencies.
+
+Supported Project Types
+-----------------------
+
+This PEP aims to serve the needs of two distinct project types:
+
+* libraries and other packages which want to define development dependencies
+
+* projects which do not build distributions which want a standardized way to
+  declare their dependency data
+
+The Python packaging toolchain is oriented towards the production of wheel and
+sdist files, containing project source code for installation. However, not all
+projects are designed or intended for this mode of distribution. Primary
+examples include webapps (which often run from source), and data science
+projects (which may start as collections of scripts and only later evolve
+package structures). This PEP seeks to offer these projects a way to specify
+their package dependencies in a way which is not constrained by the
+requirements of distribution building -- for example, a data science project
+does not need to declare a version or authors in order for these data to be
+valid, nor does it need to define an installable package source tree.
+
+Simultaneously, this PEP should benefit projects which *are* intended to build
+distributions. Because these requirements data are explicitly defined to not be
+included in any built distribution, two common use cases are addressed by this
+addition:
+
+* the use of ``extras``, as discussed above
+
+* the use of ``requirements.txt`` files for distributed libraries and
+  applications, which takes a similar form to their use in
+  non-distribution-building projects
+
+Rationale
+=========
+
+This PEP defines the storage of requirements data in lists within a
+``[dependency-groups]`` table.
+This name was chosen to matche the canonical name of the feature
+("Dependency Groups").
+
+This format should be as simple and learnable as possible, having a format
+very similar to existing ``requirements.txt`` files for many cases. Each list
+in ``[dependency-groups]`` is defined as a list of package specifiers. However,
+there are a number of use cases which require data which cannot be expressed in
+PEP 508 package specifiers. Therefore, this PEP also defines an object-format
+for package specification, capable of defining these additional data.
+
+The format does not support many forms of non-package data from
+``requirements.txt`` files, such as ``pip`` options for package servers or
+hashes. Including them in the initial stage complicates this proposal and it is
+not clear that the majority of use-cases require them. Future expansions to
+this specification may enable the use of object declarations, rather than
```

Comment:
The previous paragraph says we already have an object format. Should probably say something like "extend the object format".